### ISSUE TITLE
Flexbox layouts don't work

### ISSUE THREAD

**Issue by jdidion** [marcbachmann/node-html-pdf#24]
I am using Stylus to generate the CSS for my web page. When I enable flexbox layout (flexbox=true), nothing is rendered by node-html-pdf, but Chrome displays the page fine. When I disable flexbox layout, node-html-pdf and Chrome render the same output.

**Comment by Lichtjaeger** [comment]
Flexbox works fine if you use Phantom JS 2.0 instead of 1.9 (the standard dependency). 1.9 used an old syntax of flexbox.

**Comment by marcbachmann** [comment]
@Lichtjaeger thanks for the hint.
I'll integrate phantomjs 2 in this module as soon as it's stable on all platforms. That's also the case for the phantomjs module on npm. So we can simply switch the npm module.
Until then you can use the `phantomPath` config option and use your locally installed phantomjs 2.

**Comment by jdidion** [comment]
Yes, I can confirm that it works when I install 2.0 and set the phantomPath option to point to the new location

**Comment by ripper2hl** [comment]
@jdidion hey how to install phantomjs 2 , i cant find a npm package for this

**Comment by marcbachmann** [comment]
@ripper2hl you have to install the binary yourself. You can use `apt-get install` or some docker containers.

**Comment by ripper2hl** [comment]
Okay, thanks very much !!

**Comment by marcbachmann** [comment]
html-pdf@2.0.0 now uses phantomjs v2. Have fun

**Comment by jdidion** [comment]
Thanks for the update!

**Comment by MOUMBE** [comment]
hi flexbox ins't supported.
I use recent version of html-pdf
thanks

**Comment by sudharsanmit** [comment]
Flexbox,colors and shadows are not working . Pls find the attached input and output files.

[design_new.pdf](https://github.com/marcbachmann/node-html-pdf/files/880320/design_new.pdf)
[design_new.zip](https://github.com/marcbachmann/node-html-pdf/files/880321/design_new.zip)

**Comment by Lichtjaeger** [comment]
Yes, they broke support in version 2.1.1 in PhantomJS. ariya/phantomjs#14793

**Comment by taylorstine** [comment]
Is there a work around for this?

**Comment by sudharsanmit** [comment]
I tried so many different modules and finally gave up. Flex box and z
shadows aren't supported.

On Tuesday, April 11, 2017, Taylor Stine <notifications@github.com> wrote:
> Is there a work around for this?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.<
https://ci6.googleusercontent.com/proxy/korkejQ9CDUUjOLhuBJ4wRfk1FBEWfUF6JoSfCeHoO_3pSEHK4SXLS8EIKn4AnJ5J-t8hzhMGyQphEUfFVo-Cm196yusPXeVmc2loQasCBR0nPIzU6zk9y1Y9dk_IeFTtg6CzexR8rtb04jjCnkbBskWHuGOkw=s0-d-e1-ft#https://github.com/notifications/beacon/AP-IYWdojP09ZoD2Xt8szH0NkeLC9yWHks5ru_K3gaJpZM4EDzw3.gif>

**Comment by sun2rise** [comment]
hear they found a solution.. maybe https://github.com/ariya/phantomjs/issues/14365

**Comment by kumarharsh** [comment]
Flexbox works when you use autoprefixer `v6`, but after upgrading to v7, it broke for some reason.

**Comment by jhon-andrew** [comment]
Is there already a fix for this?

**Comment by 4muzar** [comment]
`display: flex` + `flex-wrap: wrap` works in this way:

```
display: -webkit-box
display: -webkit-flex
-webkit-flex-wrap: wrap
display: flex
flex-wrap: wrap
```
Order matters!

**Comment by alex-brain** [comment]
display: -webkit-box
display: -webkit-flex
-webkit-flex-wrap: wrap
display: flex
flex-wrap: wrap

Yes, so flex works, but justify-content doesn't work!

**Comment by givo-matan** [comment]
Hi,
The following isn't working:
    .flex-container {
      display: -webkit-box
      display: -webkit-flex
      -webkit-flex-wrap: wrap
      display: flex
      flex-wrap: wrap
    }
any ideas..?
thanks

**Comment by kumarharsh** [comment]
Use https://github.com/Szpadel/chrome-headless-render-pdf

It's based on chrome headless, and gives much better results.

**Comment by leoskyrocker** [comment]
The order property of flex didn't work and so didn't autoprefixer prefixed it for me.
After adding `-ms-flex-order` it works!

**Comment by Akash187** [comment]
For me autoprefixer are working just fine. Just use [Autoprefixer CSS Online](https://autoprefixer.github.io/) to generate css with vendor prefixes and use them. I am using inline css in a simple html file.

**Comment by gsbarreto** [comment]
Guys i find another project that works perfectly with flexbox [html-to-pdf](https://github.com/chuongtrh/html_to_pdf)

**Comment by amaneer94** [comment]
{ `flex: 1;` } is still not working for me. I am stuck. I even have tried [Autoprefixer CSS online
](https://autoprefixer.github.io/) `{
-webkit-box-flex: 1;
    -ms-flex: 1;
        flex: 1;
}
`

**Comment by soumikcv** [comment]
@amaneer94 try using  { -webkit-box-flex: 1; -webkit-flex: 1; flex: 1; } instead?

I had trouble getting justify-content to work even with autoprefixer, but based on a phantomjs issue i stumbled upon (https://github.com/ariya/phantomjs/issues/14365) adding `safari >= 4`  to my autoprefixer settings seem to fix it.

You can test it on [Autoprefixer CSS online](https://autoprefixer.github.io/ )  my inputing `safari >= 4` in the browserlist field.

**Comment by Piyush-Use-Personal** [comment]
Is anyone get around with this issue ?

**Comment by JSEvgeny** [comment]
> display: -webkit-box
> display: -webkit-flex
> -webkit-flex-wrap: wrap
> display: flex
> flex-wrap: wrap
> 
> Yes, so flex works, but justify-content doesn't work!

I got justify, align, and direction working with: `-webkit-justify-content: center`, `-webkit-align-items: center`, `-webkit-flex-flow: column`

**Comment by riandev** [comment]
display: -webkit-box;
    display: -webkit-flex;
    -webkit-flex-wrap: wrap;
    display: flex;
    flex-wrap: wrap;
    -webkit-justify-content: center;
    -webkit-align-items: center;
    -webkit-flex-flow: column;

flex not working for me anyone can help?

**Comment by dan13lJD** [comment]
Help, the bootstrap flex layout isn't work. Any solution? Any alternative?

**Comment by underscoDe** [comment]
> Help, the bootstrap flex layout isn't work. Any solution? Any alternative?

you can write your own css styles but using prefixing : `-webkit-justify-content: value;`, `-webkit-align-items: value;`, `display: -webkit-box; display: -webkit-flex;` these worked for me under ExpressJS

**Comment by MohammadK1373** [comment]
you can use this site to convert 
https://autoprefixer.github.io/
for example flex not working go to the site write {display:flex} on left box and you get  { display: -webkit-box;
display: -ms-flexbox;
display: flex;}
its really cool.
